File: 387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = [0] * 26
        for char in s:
            freq[ord(char) - ord('a')] +=1
        for i , ch in enumerate(s):
            if freq[ord(ch) - ord('a')] == 1:
                return i
        return -1

File: 387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] count = new int[26];
        char[] arr = s.toCharArray();
        for (char ch : arr) {
            count[ch - 'a']++;
        }
        for (int i=0;i<s.length();i++) {
            if (count[arr[i] - 'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
}